app/src/main/java/tech/pauly/findapet/shared/ContextProvider.java

```diff
@@ -19,16 +21,24 @@
  * this class and use the limited set of context APIs implemented in this class.
  */
 @Singleton
-public class ResourceProvider {
+public class ContextProvider {
 
     private Context context;
 
     @Inject
-    ResourceProvider(@ForApplication Context context) {
+    ContextProvider(@ForApplication Context context) {
         this.context = context;
     }
 
     public String getString(@StringRes int stringId) {
         return context.getString(stringId);
     }
+
+    public boolean hasPermission(String... permissions) {
```

Comment:
Since we made the PermissionHelper anyway, it just makes sense to put this in there instead.